tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -102,18 +107,25 @@ public void ActivityStarted<T>(string sourceName, T activity)
                     return;
                 }
 
-                ActivityMappingById.GetOrAdd(activity.Id, _ => new(activity.Instance, CreateScopeFromActivity(activity, parent, traceId, spanId, rawTraceId, rawSpanId)));
+                if (remoteActivityContext)
+                {
+                    // we need to create a new scope that maps to the remote Parent Activity ID, but uses this Activity as the instance
+                    ActivityMappingById.GetOrAdd(activity.ParentId!, _ => new(activity.Instance, CreateScopeFromActivity(activity, SpanContext.None, traceId, spanId, rawTraceId, rawSpanId)));
```

Comment:
What's the reasoning for storing via the parent ID? I would think you need to store the activity by its current ID so that any child spans of this span can lookup their parent (aka this span)

```suggestion
                    ActivityMappingById.GetOrAdd(activity.Id, _ => new(activity.Instance, CreateScopeFromActivity(activity, SpanContext.None, traceId, spanId, rawTraceId, rawSpanId)));
```

Reply:
Ah yep good catch, ParentId was a leftover from a earlier attempt of mine, but isn't correct anymore, I'll have to also update the `SpanContext.None` here to a new one for your other comment as well.